src/web/mod.rs

```diff
@@ -581,7 +599,13 @@ mod test {
 
     fn version(v: Option<&str>, db: &TestDatabase) -> Option<String> {
         match_version(&mut db.conn(), "foo", v)
-            .and_then(|version| version.assume_exact().map(|semver| semver.into_parts().0))
+            .ok()
+            .and_then(|version| {
+                version
+                    .assume_exact()
+                    .ok()
+                    .map(|semver| semver.into_parts().0)
+            })
```

Comment:
nit: this would be easier to read with `?`:
```suggestion
         match_version(&mut db.conn(), "foo", v).ok()?.assume_exact().ok()?.into_parts().0
```